crates/evm/evm/src/executors/invariant/error.rs

```diff
@@ -296,153 +264,39 @@ impl FailedInvariantCaseData {
-        executor: &Executor,
-        runs: usize,
-        retries: usize,
-    ) -> eyre::Result<Vec<usize>> {
-        // Construct a ArcRwLock vector of indices of `calls`
-        let shrunk_call_indices = Arc::new(RwLock::new((0..calls.len()).collect()));
-        let shrink_limit = self.shrink_run_limit - runs;
-
-        let upper_bound = calls.len().saturating_sub(1);
-        // We construct either a full powerset (this guarantees we maximally shrunk for the given
-        // calls) or a random subset
-        let (set_of_indices, is_powerset): (Vec<_>, bool) = if calls.len() <= 64 &&
-            (1 << calls.len() as u32) <= shrink_limit
-        {
-            // We add the last tx always because thats ultimately what broke the invariant
-            let powerset = (0..upper_bound)
-                .powerset()
-                .map(|mut subset| {
-                    subset.push(upper_bound);
-                    subset
-                })
-                .collect();
-            (powerset, true)
-        } else {
-            // construct a random set of subsequences
-            let mut rng = thread_rng();
-            (
-                (0..shrink_limit / 3)
-                    .map(|_| {
-                        // Select between 1 and calls.len() - 2 number of indices
-                        let amt: usize = rng.gen_range(1..upper_bound);
-                        // Construct a random sequence of indices, up to calls.len() - 1 (sample is
-                        // exclusive range and we dont include the last tx
-                        // because its always included), and amt number of indices
-                        let mut seq = seq::index::sample(&mut rng, upper_bound, amt).into_vec();
-                        // Sort the indices because seq::index::sample is unordered
-                        seq.sort();
-                        // We add the last tx always because thats what ultimately broke the
-                        // invariant
-                        seq.push(upper_bound);
-                        seq
-                    })
-                    .collect(),
+        sequence: Vec<usize>,
+    ) -> bool {
+        // Apply the shrinked candidate sequence.
+        sequence.iter().for_each(|call_index| {
+            let (sender, (addr, bytes)) = &calls[*call_index];
+            executor.call_raw_committing(*sender, *addr, bytes.clone(), U256::ZERO).unwrap();
+        });
+
+        // Check the invariant for candidate sequence.
+        // If sequence fails then we can continue with shrinking - the removed call does not affect
+        // failure.
+        //
+        // If sequence doesn't fail then we have to restore last removed call and continue with next
+        // call - removed call is a required step for reproducing the failure.
+        if let Some(func) = &self.func {
+            let mut call_result =
+                executor.call_raw(CALLER, self.addr, func.clone(), U256::ZERO).unwrap();
```

Comment:
same here for the error

Reply:
yep, handled in the same follow up PR as before where `func` is not an option anymore https://github.com/grandizzy/foundry/commit/7220ea7ff70db0ed9f532345c1e4137e3171ba4b#diff-a894285abe6ea94aaa1b3bfe177642202d9c46a5d8e244b54ec0a53764acb868R292-R299